python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -83,7 +102,20 @@ def execute(self, taskDef, quantum, butler):
                       f"task={taskClass.__name__} dataId={quantum.dataId}.")
             return
 
-        quantum = self.updatedQuantumInputs(quantum, butler)
+        try:
+            quantum = self.updatedQuantumInputs(quantum, butler, taskDef)
+        except NoWorkFound as exc:
+            _LOG.info("Nothing to do for task '%s' on quantum %s; skipping: %s",
+                      taskDef.label, quantum.dataId, str(exc))
+            # Make empty metadata that looks something like what a do-nothing
+            # task would write (but we don't bother with empty nested
+            # PropertySets for subtasks).  This is slightly duplicative with
+            # logic in pipe_base that we can't easily call from here; we'll fix
+            # this on DM-29761.
+            fullMetadata = PropertySet()
+            fullMetadata.set(taskDef.label, PropertyList())
```

Comment:
Please don't use `set` with a `PropertyX` any more. This should all work with:

```
fullMetadata[taskeDef.label] = {}
fullMetadata["quantum"] = quantumMetadata
```
and the `__setitem__` code automatically fixes things up internally.  That way when PropertySet is removed in the line above nothing else needs to be edited.

Reply:
Looking at https://lsst.ncsa.illinois.edu/~yusra/download/log.log maybe this needs to reraise its own `NoWorkFound`?

Reply:
Yeah, it needs to do something there, but I think `return` early is what we want.